tests/kafkatest/tests/core/replication_test.py

```diff
@@ -133,7 +133,11 @@ def min_cluster_size(self):
     @parametrize(failure_mode="hard_bounce",
             broker_type="leader",
             security_protocol="SASL_SSL", client_sasl_mechanism="SCRAM-SHA-256", interbroker_sasl_mechanism="SCRAM-SHA-512")
-    def test_replication_with_broker_failure(self, failure_mode, security_protocol, broker_type, client_sasl_mechanism="GSSAPI", interbroker_sasl_mechanism="GSSAPI"):
+    @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
+            security_protocol=["PLAINTEXT"], broker_type=["leader"], compression_type=["gzip"])
+    def test_replication_with_broker_failure(self, failure_mode, security_protocol, broker_type,
+                                             client_sasl_mechanism="GSSAPI", interbroker_sasl_mechanism="GSSAPI",
+                                             compression_type=None):
```

Comment:
It will continue to run the current parameterizations plus four new test cases which exercise gzip compression.